processor/stream/processor.go

```diff
@@ -50,30 +50,35 @@ const (
 
 type decodeMetadataFunc func(decoder.Decoder, *model.APMEvent) error
 
+// Processor decodes a stream
 type Processor struct {
-	MaxEventSize     int
 	streamReaderPool sync.Pool
 	decodeMetadata   decodeMetadataFunc
+	sem              chan struct{}
+	MaxEventSize     int
 }
 
-func BackendProcessor(cfg *config.Config) *Processor {
+func BackendProcessor(cfg *config.Config, sem chan struct{}) *Processor {
```

Comment:
Should we create the channel internally using the setting in `cfg`, rather than passing one in?

Reply:
First, I created a semaphore for each of the backend processors, but that isn't a great solution because each of the stream processors (Backend, RUM, RUMV3) would have its own limit, then the calculation doesn't hold anymore or we need to divide it by 3 which severely would limit the upper bound of concurrent decodes that a single backend processor performs.

Instead, if we choose to have an "intake semaphore" then the limits are shared between all the ingesting processors.

Reply:
Right, makes sense.